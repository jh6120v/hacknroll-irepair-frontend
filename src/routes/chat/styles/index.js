import styled, { css } from 'styled-components';
import camera from '../../../assets/images/camera.svg';
import sent from '../../../assets/images/sent.svg';
import tailLeft from '../../../assets/images/tail-left.svg';
import tailRight from '../../../assets/images/tail-right.svg';

const ChatContainer = styled.div`
  display: flex;
  flex-wrap: nowrap;
  align-content: flex-start;
  flex-direction: column;
  background-color: #fff;
  box-shadow: 0 5px 18px 0 rgba(200,75,98,1);
  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
  //margin-top: 13px;
  height: calc(100vh - 44px - env(safe-area-inset-top));
  overflow: hidden;
  overflow-scrolling: touch;
`;

const ChatContent = styled.div`
  height: 77vh;
  padding: 29px 15px 0 15px;
  flex: 1 1 auto;
  display: flex;
  flex-wrap: wrap;
  overflow: auto;
  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
  align-content: flex-start;
`;

const ChatBody = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-wrap: wrap;
  overflow-y: auto;
  overflow-x: hidden;
`;

const ChatBoxContainer = styled.div`
  display: flex;
  width: 100%;
  margin-bottom: 13px;
  align-self: flex-start;
  ${(props) => {
        if (props.theme.isSelf) {
            return css`
          flex-direction: row-reverse;
        `;
        }
    }}
`;

const ChatBoxContent = styled.div`
  display: flex;
  max-width: 80%;
  flex-wrap: nowrap;
  padding: ${(props) => (props.theme.isSelf ? '0 0 0 5px' : '0 5px 0 0')};
`;

const Time = styled.div`
  max-width: 20%;
  display: flex;
  align-items: flex-end;
  color: #f1828d;
  font-size: 11px;
  letter-spacing: .21px;
  font-weight: 300;
`;

const Avatar = styled.div`
  width: 47px;
  height: 47px;
  border-radius: 50%;
  ${(props) => (css`background-image: url(${props.image})`)};
  background-repeat: no-repeat;
  background-size: cover;
`;

const Dialog = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: ${(props) => (props.theme.isSelf ? '100%' : 'calc(100% - 58px)')};
  ${(props) => {
        if (props.theme.isSelf) {
            return css`
              width: 100%;
              margin-right: 5px;
            `;
        }

        return css`
          width: calc(100% - 58px);
          margin-left: 11px;
        `;
    }};
`;

const Author = styled.div`
  width: 100%;
  font-size: 13px;
  line-height: 25px;
  letter-spacing: .25px;
  color: #f1828d;
  font-weight: 300;
`;

const Bubble = styled.div`
  position: relative;
  width: 100%;
  min-height: 37px;
  border-radius: 18px;
  font-size: 16px;
  line-height: 21px;
  z-index: 1;
  white-space: pre-line;
  ${(props) => {
        if (props.theme.isSelf) {
            return css`
          padding: 8px 12px 8px 11px;
          margin-right: 5px;
          color: #fff;
          background-color: #f1828d;
        `;
        }
        return css`
          padding: 8px 11px 8px 12px;
          color: #1a1a1a;
          background-color: #f1f2f3;
        `;
    }};
  &::after {
    content: '';
    position: absolute;
    ${(props) => {
        if (props.theme.isSelf) {
            return css`
              right: -7px;
              top: -8px;
              background-image: url(${tailRight});
              background-position: right top;
            `;
        }
        return css`
              left: -6px;
              top: -1px;
              background-image: url(${tailLeft});
              background-position: left top;
            `;
    }};
    width: 21px;
    height: 25px;
    background-repeat: no-repeat;
    z-index: -1;
  }
`;

const Img = styled.div`
  width: 100%;
  & > img {
    max-width: 100%;
    height: auto;
  }
`;

const ChatBar = styled.div`
  display: flex;
  width: 100%;
  background-color: #f1828d;
  z-index: 3;
  padding: 0 11px;  
`;

const ChatBarInner = styled.div`
  display: flex;
  width: 100%;
  align-items: flex-end;
  justify-content: space-around;
  align-items: center;
  margin-top: 15px;
  margin-bottom: ${(props) => props.toggleKeyboard || props.safeArea.sab === '0px' ? '18px' : 'env(safe-area-inset-bottom)'};
`;

const Camera = styled.div`
  flex: 1 1 auto;
  width: 37px;
  height: 37px;
  background-image: url(${camera});
  background-repeat: no-repeat;
  background-position: center center;
  background-size: 31px 31px;
  overflow: hidden;
  & > input {
    opacity: 0;
    height: 100%;
  }
`;

const TypingBar = styled.div`
  display: flex;
  flex-direction: column;
  width: calc(100% - 90px);
  overflow: hidden;
  background-color: #fff;
  border-radius: 17.5px;
  > textarea {
    width: 100%;
    padding: 8px 15px;
    color: #f1828d;
    font-size: 15px;
    line-height: 17px;
  }
  > ::placeholder {
    color: #f1828d;
    text-align: center;
  }
`;

const Sent = styled.div`
  flex: 1 1 auto;
  width: 37px;
  height: 37px;
  background-image: url(${sent});
  background-repeat: no-repeat;
  background-position: center center;
  background-size: 31px 31px;
`;

export {
    ChatContainer,
    ChatContent,
    ChatBoxContainer,
    ChatBoxContent,
    ChatBody,
    Time,
    Avatar,
    Dialog,
    Author,
    Bubble,
    Img,
    ChatBar,
    ChatBarInner,
    Camera,
    TypingBar,
    Sent
};
