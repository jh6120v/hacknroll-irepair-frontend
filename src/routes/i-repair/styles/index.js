import styled, { css } from 'styled-components';
import icon1 from '../../../assets/images/btn-icon-1.svg';
import icon2 from '../../../assets/images/btn-icon-2.svg';
import icon3 from '../../../assets/images/btn-icon-3.svg';

const Panel = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-content: flex-start;
  height: 66vh;
  padding-bottom: env(safe-area-inset-bottom);
  background-color: #fff;
  box-shadow: 0 5px 18px 0 rgba(200,75,98,1);
  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
  z-index: 1;
  ${(props) => (props.absolute ? css`
    position: absolute;
    left: 0;
    bottom: 0;
  ` : null)}
`;

const PanelTitle = styled.div`
  width: 100%;
  font-size: 23px;
  color: #f1828d;
  padding: 68px 0 35px 0;
  text-align: center;
`;

const PanelButtonWrap = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-content: flex-start;
`;

const PanelButton = styled.div`
  width: 67%;
  border-radius: 10px;
  font-size: 19px;
  letter-spacing: .68px;
  padding: 22px 0;
  margin-bottom: 20px;
  color: #fff;
  text-align: center;
  font-weight: 300;
  background-color: ${(props) => (props.bgColor ? props.bgColor : '#f1828d')};
  ${(props) => {
        if (props.icon) {
            let icon;
            switch (props.icon) {
                case 'icon-1':
                    icon = icon1;
                    break;
                case 'icon-2':
                    icon = icon2;
                    break;
                case 'icon-3':
                default:
                    icon = icon3;
            }

            return css`
              background-repeat: no-repeat;
              background-size: 28px 28px;
              background-position: 34px center;
              background-image: url(${icon});
            `;
        }
    }}
`;

export {
    Panel,
    PanelTitle,
    PanelButtonWrap,
    PanelButton
};
