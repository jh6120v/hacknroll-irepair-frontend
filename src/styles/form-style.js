import styled from 'styled-components';

const Form = styled.form`
  width: 100%;
  padding: 0 10%;
`;

const FormGroup = styled.div`
  width: 100%;
  margin-bottom: 20px;
`;

const InputGroup = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  width: 100%;
  height: 55px;
  border-radius: 5px;
  overflow: hidden;
  margin-bottom: 20px;
  background-color: #fff;
`;

const InputText = styled.input`
  width: 100%;
  height: 35px;
  line-height: 35px;
  font-size: 18px;
  letter-spacing: .3px;
  color: #f1828d;
  text-align: center;
  border-width: 0;
  -webkit-text-fill-color: #f1828d;
  &::placeholder {
    color: #f1828d;
    text-align: center;
  }
`;

const ValidateErrorMsg = styled.div`
  width: 100%;
  color: #fff;
  text-align: center;
  padding-top: 10px;
`;

const ButtonGroup = styled(FormGroup)``;

const SubmitBtn = styled.button`
  width: 100%;
  height: 55px;
  font-size: 20px;
  letter-spacing: .33px;
  color: #fff;
  line-height: 55px;
  text-align: center;
  background-color: #ff6f89;
  box-shadow: 0 3px 15px 0 #d06a74;
  border: none;
  padding: 0;
`;

export {
    Form,
    FormGroup,
    InputGroup,
    InputText,
    ValidateErrorMsg,
    ButtonGroup,
    SubmitBtn
};
