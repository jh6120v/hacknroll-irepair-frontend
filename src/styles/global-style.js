import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html {
    padding-top: env(safe-area-inset-top);
    padding-right: env(safe-area-inset-right);
    padding-bottom: env(safe-area-inset-bottom);
    padding-left: env(safe-area-inset-left);
  }
  html, body {
    height: auto;
    font-size: 16px;
    font-family: 'Noto Sans TC', sans-serif;
    color: #1d1d1f;
    -webkit-tap-highlight-color: transparent;
    -webkit-touch-callout: none;
    -webkit-user-select: none;
    user-select: none;
    background: #ff6f89;
    background: -moz-linear-gradient(-45deg,  #ff6f89 0%, #ff9478 100%);
    background: -webkit-linear-gradient(-45deg,  #ff6f89 0%,#ff9478 100%);
    background: linear-gradient(135deg,  #ff6f89 0%,#ff9478 100%);
    filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#ff6f89', endColorstr='#ff9478',GradientType=1 );
  }
  #app {
    display: flex;
    height: 100%;
  }
  a {
    text-decoration: none;
    cursor: pointer;
  }
`;

export default GlobalStyle;
