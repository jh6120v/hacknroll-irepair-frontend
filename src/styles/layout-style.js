import styled from 'styled-components';
import { respondTo } from './_mixin';
import logo from '../assets/images/i-repair-logo.svg';

const Header = styled.div`
  display: flex;
  justify-content: center;
  height: 34vh;
  padding: 11vh 0 5vh 0;
`;

const Logo = styled.div`
  width: 100%;
  background-image: url(${logo});
  background-position: center center;
  background-size: contain;
  background-repeat: no-repeat;
`;

const Wrapper = styled.div`
  display: flex;
  max-width: 400px;
  min-height: 100%;
  flex-wrap: wrap;
  flex-direction: column;
  margin: 0;
  ${respondTo.xs`
    margin: 0 auto;
  `}
`;

const Container = styled.div`
  width: 100%;
  height: 100%;
  flex-grow: 1;
  flex-shrink: 1;
  flex-basis: auto;
`;

const ContainerInner = styled.div`
  width: 100%;
  position: relative;
  background: #ff6f89;
  background: -moz-linear-gradient(-45deg,  #ff6f89 0%, #ff9478 100%);
  background: -webkit-linear-gradient(-45deg,  #ff6f89 0%,#ff9478 100%);
  background: linear-gradient(135deg,  #ff6f89 0%,#ff9478 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#ff6f89', endColorstr='#ff9478',GradientType=1 );
`;

export {
    Header,
    Logo,
    Wrapper,
    Container,
    ContainerInner
};
