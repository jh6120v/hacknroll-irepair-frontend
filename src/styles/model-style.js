import styled from 'styled-components';

const ModelShadow = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  background-color: rgba(4,4,15,.4);
  z-index: 10;
`;

const ModelWrap = styled.div`
  position: fixed;
  left: 50%;
  top: 50%;
  width: 270px;
  max-width: 72%;
  background-color: rgb(248, 248, 248);
  z-index: 11;
  display: flex;
  flex-direction: column;
  transform: translateX(-50%) translateY(-50%);
  border-radius: 14px;
`;

const ModelTitle = styled.div`
  width: 100%;
  padding: 19px 16px 0 16px;
  color: #000;
  font-size: 17px;
  text-align: center;
  letter-spacing: -.41px;
  line-height: 22px;
`;

const ModelContent = styled.div`
  width: 100%;
  padding: 0 16px 24px 16px;
  color: #000;
  font-size: 13px;
  text-align: center;
  line-height: 18px;
  letter-spacing: -.08px;
  font-weight: 300;
`;

const ModelFooter = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: nowrap;
  border-top: .5px solid rgba(4, 4, 15, .4);
`;

const Button = styled.div`
  width: 100%;
  text-align: center;
  font-size: 17px;
  line-height: 22px;
  letter-spacing: -.41px;
  border-right: .5px solid rgba(4, 4, 15, .4);
  padding: 11.5px 0 10px 0;
  &:last-child:not(:first-child) {
    border-right: none;
  }
  color: ${(props) => props.type === 'confirm' ? '#007aff' : '#8e8e93'};
`;

export {
    ModelShadow,
    ModelWrap,
    ModelTitle,
    ModelContent,
    ModelFooter,
    Button
};
