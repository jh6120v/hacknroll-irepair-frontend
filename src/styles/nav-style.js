import styled from 'styled-components';

const Nav = styled.div`
  width: 100%;
  height: calc(44px + env(safe-area-inset-top));
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: #fff;
  padding: env(safe-area-inset-top) 10px 0 10px;
  overflow: hidden;
`;

const Title = styled.div`
  font-size: 17px;
  font-weight: bold;
  line-height: 22px;
`;

const PrevLink = styled.div`
  min-width: 28px;
  height: 28px;
`;

const NextLink = styled(PrevLink)``;

export {
    Nav,
    Title,
    PrevLink,
    NextLink,
};
