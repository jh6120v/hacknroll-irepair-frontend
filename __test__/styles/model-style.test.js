import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { Button, ModelContent, ModelFooter, ModelShadow, ModelTitle, ModelWrap } from '../../src/styles/model-style';

describe('test model-style', () => {
    it('test <ModelShadow /> snapshot.', () => {
        const { container } = render(<ModelShadow />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ModelShadow /> when render it.', () => {
        const { container } = render(<ModelShadow />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('position', 'fixed');
        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('height', '100%');
        expect(tree).toHaveStyleRule('left', '0');
        expect(tree).toHaveStyleRule('top', '0');
        expect(tree).toHaveStyleRule('background-color', 'rgba(4,4,15,.4)');
        expect(tree).toHaveStyleRule('z-index', '10');
    });

    it('test <ModelWrap /> snapshot.', () => {
        const { container } = render(<ModelWrap />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ModelWrap /> when render it.', () => {
        const { container } = render(<ModelWrap />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('position', 'fixed');
        expect(tree).toHaveStyleRule('left', '50%');
        expect(tree).toHaveStyleRule('top', '50%');
        expect(tree).toHaveStyleRule('width', '270px');
        expect(tree).toHaveStyleRule('max-width', '72%');
        expect(tree).toHaveStyleRule('background-color', 'rgb(248,248,248)');
        expect(tree).toHaveStyleRule('z-index', '11');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
        expect(tree).toHaveStyleRule('transform', 'translateX(-50%) translateY(-50%)');
        expect(tree).toHaveStyleRule('border-radius', '14px');
    });

    it('test <ModelTitle /> when render it.', () => {
        const { container } = render(<ModelTitle />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('padding', '19px 16px 0 16px');
        expect(tree).toHaveStyleRule('color', '#000');
        expect(tree).toHaveStyleRule('font-size', '17px');
        expect(tree).toHaveStyleRule('text-align', 'center');
        expect(tree).toHaveStyleRule('letter-spacing', '-.41px');
        expect(tree).toHaveStyleRule('line-height', '22px');
    });

    it('test <ModelContent /> snapshot.', () => {
        const { container } = render(<ModelContent />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ModelContent /> when render it.', () => {
        const { container } = render(<ModelContent />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('padding', '0 16px 24px 16px');
        expect(tree).toHaveStyleRule('color', '#000');
        expect(tree).toHaveStyleRule('font-size', '13px');
        expect(tree).toHaveStyleRule('text-align', 'center');
        expect(tree).toHaveStyleRule('line-height', '18px');
        expect(tree).toHaveStyleRule('letter-spacing', '-.08px');
        expect(tree).toHaveStyleRule('font-weight', '300');
    });

    it('test <ModelFooter /> snapshot.', () => {
        const { container } = render(<ModelFooter />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ModelFooter /> when render it.', () => {
        const { container } = render(<ModelFooter />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-wrap', 'nowrap');
        expect(tree).toHaveStyleRule('border-top', '.5px solid rgba(4,4,15,.4)');
    });

    it('test <Button /> when render it.', () => {
        const { container } = render(<Button />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('text-align', 'center');
        expect(tree).toHaveStyleRule('font-size', '17px');
        expect(tree).toHaveStyleRule('line-height', '22px');
        expect(tree).toHaveStyleRule('letter-spacing', '-.41px');
        expect(tree).toHaveStyleRule('border-right', '.5px solid rgba(4,4,15,.4)');
        expect(tree).toHaveStyleRule('padding', '11.5px 0 10px 0');
        expect(tree).toHaveStyleRule('border-right', 'none', {
            modifier: '&:last-child:not(:first-child)'
        });
        expect(tree).toHaveStyleRule('color', '#8e8e93');
    });

    it('test <Button> when render prop type confirm.', () => {
        const { container } = render(<Button type="confirm" />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('color', '#007aff');
    });
});
