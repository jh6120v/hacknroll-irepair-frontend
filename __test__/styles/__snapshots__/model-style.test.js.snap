// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test model-style test <ModelContent /> snapshot. 1`] = `
.c0 {
  width: 100%;
  padding: 0 16px 24px 16px;
  color: #000;
  font-size: 13px;
  text-align: center;
  line-height: 18px;
  -webkit-letter-spacing: -.08px;
  -moz-letter-spacing: -.08px;
  -ms-letter-spacing: -.08px;
  letter-spacing: -.08px;
  font-weight: 300;
}

<div
  class="c0"
/>
`;

exports[`test model-style test <ModelFooter /> snapshot. 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  border-top: .5px solid rgba(4,4,15,.4);
}

<div
  class="c0"
/>
`;

exports[`test model-style test <ModelShadow /> snapshot. 1`] = `
.c0 {
  position: fixed;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  background-color: rgba(4,4,15,.4);
  z-index: 10;
}

<div
  class="c0"
/>
`;

exports[`test model-style test <ModelWrap /> snapshot. 1`] = `
.c0 {
  position: fixed;
  left: 50%;
  top: 50%;
  width: 270px;
  max-width: 72%;
  background-color: rgb(248,248,248);
  z-index: 11;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
  border-radius: 14px;
}

<div
  class="c0"
/>
`;
